### PR TITLE
Switch to sidekiq.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
sidekiq++

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test) that do accessioning and/or create_preassembly_image_spec as it tests full preservation*** and/or test in stage environment, in addition to specs. ⚡

### CODE REVIEW

File: config/sidekiq.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+:concurrency: 5
+:queues:
+  - preservationIngestWF_default
+  - preservationIngestWF_low
```
Comment (ndushay) [inline config/sidekiq.yml:5]: why is prescat the only set of robots with this?

File: spec/preservation_ingest/update_moab_spec.rb [hunk 2]
```diff
@@ -33,11 +32,9 @@
         allow(mock_new_version).to receive(:verify_version_storage).and_return(verification_result)
         allow(verification_result).to receive(:to_json).and_return('some_json')
         allow(verification_result).to receive(:entity).and_return(full_druid_vers)
-        allow(LyberCore::Log).to receive(:info).with('some_json')
```
Comment (ndushay) [inline spec/preservation_ingest/update_moab_spec.rb:36]: you've decided not to test for logging?  (just checking)

**Review by ndushay — APPROVED** [review]
We have tonight, for your reviewing pleasure, zee comments du this PR, but please do not feel you are obligated to agree with them before merging.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 require 'simplecov'
-require 'byebug'
 
 SimpleCov.start do
   add_filter 'spec'
   add_filter 'config'
 end
 
-require File.join(File.dirname(__FILE__), '..', 'config/boot')
+ENV['ROBOT_ENVIRONMENT'] = 'test'
```
Comment (ndushay) [inline spec/spec_helper.rb:10]: do we still need this, if we got rid of the config/environments?

**Review by jmartin-sul — APPROVED** [review]
a couple very tiny nitpicks to consider, but definitely no objection to merging

File: config/sidekiq.yml (hunk 1 above)
Comment (jmartin-sul) [reply config/sidekiq.yml:5]: we used to configure similarly for common-accessioning, but from looking at that PR, seems like its `config/sidekiq.yml` did get moved to shared_configs?

if it doesn't change regularly or differ between environments, i think i'm personally indifferent?

Comment (justinlittman) [reply config/sidekiq.yml:5]: If it is the same across all workers for all environments, there is no need to put in shared configs. (Though one might make the case for putting it there anyway for consistency.)

File: spec/preservation_ingest/update_moab_spec.rb (hunk 2 above)
Comment (justinlittman) [reply spec/preservation_ingest/update_moab_spec.rb:36]: Yes, unless the logging is critical for some reason, I think checking it unnecessarily bloats tests.

File: spec/spec_helper.rb (hunk 3 above)
Comment (justinlittman) [reply spec/spec_helper.rb:10]: Yes, it is required to set up `config/settings`.

File: config/deploy/prod.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
-server 'preservation-robots1-prod.stanford.edu', user: 'pres', roles: %w[web app db monitor]
+server 'preservation-robots1-prod.stanford.edu', user: 'pres', roles: %w[web app db monitor worker]
 # # robots2-prod is a warm standby - no resque-pool up but ready to deploy to if needed
-# server 'preservation-robots2-prod.stanford.edu', user: 'pres', roles: %w[web app db monitor]
+# server 'preservation-robots2-prod.stanford.edu', user: 'pres', roles: %w[web app db monitor worker]
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply config/deploy/prod.rb:5]: In theory, you're right. However, sometimes critical deploy steps are attached to unexpected roles. I don't really want to test this, so I'm going to leave `db` role as is and put back where I deleted it.

File: config/sidekiq.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply config/sidekiq.yml:1]: Yeah, I'm not sure Julian's comment will really match the new configuration so I'd say lets see how it goes.

File: config/sidekiq.yml (hunk 1 above)
Comment (ndushay) [reply config/sidekiq.yml:5]: ah - yes, I like it here in the robot code if it's the same everywhere - but having it in both places is fine, too.

File: config/sidekiq.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply config/sidekiq.yml:1]: wfm

File: config/deploy/prod.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply config/deploy/prod.rb:5]: good point!

**Pull request merged**